### ISSUE TITLE
azure-cli keeps trying to install azure-devops extension despite it being installed (when --organization parameter is in a specific place)

### ISSUE THREAD

**Issue by myhau** [Azure/azure-cli#17688]
## Describe the bug

**Command Name**
All commands related to azure devops / pipelines (e.g. `az devops`) where **--organization parameter is after the command (not after subcommands)**.

For example, this happens for `az devops --organization '{org}' project list`, but not for `az devops project list --organization '{org}'`.

**Errors:**

Command keeps printing this message and trying to install `azure-devops` extension, despite it being already installed:

```
The command requires the extension azure-devops. Do you want to install it now? The command will continue to run after the extension is installed. (Y/n): Y
```

Entering `Y` repeats this process:

```
az devops --organization tesco-location projects list
The command requires the extension azure-devops. Do you want to install it now? The command will continue to run after the extension is installed. (Y/n): Y
Run 'az config set extension.use_dynamic_install=yes_without_prompt' to allow installing extensions without prompt.
Extension 'azure-devops' 0.18.0 is already installed.
It will be updated if available.
No updates available for 'azure-devops'. Use --debug for more information.
The command requires the extension azure-devops. Do you want to install it now? The command will continue to run after the extension is installed. (Y/n): Y
Run 'az config set extension.use_dynamic_install=yes_without_prompt' to allow installing extensions without prompt.
Extension 'azure-devops' 0.18.0 is already installed.
It will be updated if available.
No updates available for 'azure-devops'. Use --debug for more information.
The command requires the extension azure-devops. Do you want to install it now? The command will continue to run after the extension is installed. (Y/n): Y
Run 'az config set extension.use_dynamic_install=yes_without_prompt' to allow installing extensions without prompt.
Extension 'azure-devops' 0.18.0 is already installed.
It will be updated if available.
No updates available for 'azure-devops'. Use --debug for more information.
The command requires the extension azure-devops. Do you want to install it now? The command will continue to run after the extension is installed. (Y/n):
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- `az devops --organization {your-organization} list projects`
- enter `Y` when asked for extension installation

## Environment Summary
```
macOS-11.2.3-x86_64-i386-64bit
Python 3.8.9
Installer: HOMEBREW

azure-cli 2.22.0

Extensions:
azure-devops 0.18.0

```

**Comment by yungezz** [comment]
hi @fengzhou-msft, could you look is this a package issue or devops extension specific issue? thanks

**Comment by fengzhou-msft** [comment]
Fixed in version `2.22.1`.
